code/modules/clothing/masks/miscellaneous.dm

```diff
@@ -120,6 +120,71 @@
 	mute = MUTE_NONE
 	security_lock = TRUE
 	locked = FALSE
+	materials = list(MAT_METAL=500, MAT_GLASS=50)
+
+/obj/item/clothing/mask/muzzle/safety/shock
+	name = "shock muzzle"
+	desc = "A muzzle designed to prevent biting.  This one is fitted with a behavior correction system."
+	var/obj/item/assembly/trigger = null
+	origin_tech = "materials=1;engineering=1"
+	materials = list(MAT_METAL=500, MAT_GLASS=50)
+
+/obj/item/clothing/mask/muzzle/safety/shock/attackby(obj/item/W, mob/user, params)
+	if(isscrewdriver(W))
+		to_chat(user, "<span class='notice'>You disassemble [src].</span>")
+		trigger.forceMove(get_turf(user))
+		trigger.master = null
+		trigger.holder = null
+		trigger = null
+	else if(trigger)

```

Comment:
I'm a little nervous about `trigger` preventing any parent calls here - would it be better to do this check inside the assembly attachment clause?

Reply:
parent calls will now pass through unless the attackby is an assembly component